lib/core/stores/local.ts

```diff
@@ -209,4 +265,101 @@ export class LocalPromiseStore implements IPromiseStore {
   ): AsyncGenerator<DurablePromise[], void> {
     return this.storage.search(id, state, tags, limit);
   }
+
+  async createSchedule(
+    id: string,
+    ikey: string | undefined,
+    description: string | undefined,
+    cron: string,
+    tags: Record<string, string> | undefined,
+    promiseId: string,
+    promiseTimeout: number,
+    promiseHeaders: Record<string, string>,
+    promiseData: string | undefined,
+    promiseTags: Record<string, string> | undefined,
+  ): Promise<Schedule> {
+    const schedule = {
+      id,
+      description,
+      cron,
+      tags,
+      promiseId,
+      promiseTimeout,
+      promiseParam: {
+        headers: promiseHeaders,
+        data: promiseData,
+      },
+      promiseTags,
+      lastRunTime: undefined,
+      nextRunTime: this.calculateNextRunTime(cron),
+      idempotencyKey: ikey,
+      createdOn: Date.now(),
+    };
+
+    this.schedules.push(schedule);
+
+    return schedule;
+  }
+
+  async deleteSchedule(id: string): Promise<boolean> {
+    const index = this.schedules.findIndex((schedule) => schedule.id === id);
+
+    if (index !== -1) {
+      this.schedules.splice(index, 1);
+    }
+
+    return true;
+  }
+
+  async getSchedule(id: string): Promise<Schedule> {
+    const schedule = this.schedules.find((schedule) => schedule.id === id);
+
+    if (schedule) {
+      return schedule;
+    }
+
+    throw new ResonateError(ErrorCodes.NOT_FOUND, "Not found");
+  }
+  async searchSchedules(
+    id: string,
+    tags?: Record<string, string>,
+    limit?: number,
+    cursor?: string | undefined,
+  ): Promise<{ cursor: string; schedules: Schedule[] }> {
+    let filteredSchedules = this.schedules.filter((schedule) => schedule.id === id);
+
+    if (tags) {
+      filteredSchedules = filteredSchedules.filter((schedule) =>
+        Object.entries(tags).every(([key, value]) => schedule.tags && schedule.tags[key] === value),
+      );
+    }
+
+    if (cursor !== undefined) {
+      const cursorIndex = filteredSchedules.findIndex((schedule) => schedule.id === cursor);
+      if (cursorIndex !== -1) {
+        filteredSchedules = filteredSchedules.slice(cursorIndex + 1);
+      }
+    }
+
+    if (limit !== undefined) {
+      filteredSchedules = filteredSchedules.slice(0, limit);
+    }
+
+    const nextCursor = filteredSchedules.length > 0 ? filteredSchedules[filteredSchedules.length - 1].id : null;
+
+    return {
+      cursor: nextCursor ?? "",
+      schedules: filteredSchedules,
+    };
+  }
+
+  private calculateNextRunTime(cron: string): number | undefined {
+    try {
+      const nextRunDate = cronParser.parseExpression(cron).next();
+      return nextRunDate.getTime();
+    } catch (error) {
+      console.error("Error calculating next run time:", error);
+      return undefined;
```

Comment:
We're going to want to throw an error rather than swallow it, can we also split up parse and validate?